cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -828,6 +788,38 @@ private boolean canRoll(int nodeId, boolean isController, boolean isBroker, long
         }
     }
 
+    private boolean canRollController(NodeRef nodeRef, long timeout, TimeUnit unit) throws UnforceableProblem, InterruptedException, ForceableProblem {
+        // Discover the active controller first so that the admin client can be bootstrapped only with the active controller.
+        // This is because quorum info can be requested only from the active controller.
+        // TODO: Once KAFKA-18230 is completed and supported in Strimzi, we can remove this method,
+        //       and simply use the regular controllerAdminClient for the quorum check
+        int activeController = getActiveController(nodeRef, controllerAdminClient, operationTimeoutMs, TimeUnit.MILLISECONDS);
+        if (activeController < 0) {
+            // TODO: if the active controller is returned as -1 (maybe election was still in progress),
+            //      use the admin client bootstrapped with all controller nodes to see if it can hit the active controller by luck.
+            //      If it does not hit the active controller, it will return an error and this node will be retried later anyway.
+            //      Otherwise we can just return false here, and the node will be then retried later.
```

Comment:
This could cause the Warning message with NOT_LEADER_OR_FOLLOWER but both would result in retrying the node if active controller is not found.

Reply:
What exactly are the situations when we expect to run into this? When there are no running brokers and only controllers?

Reply:
We are querying the active controller from the controllers, not brokers so no running brokers should not be a problem. I think they could return -1, if there is np leader elected yet/election is still in progress? Could be also an indication of quorum being in a bad state.

Reply:
If we cannot get the active controller (for any reasons) how much is it dangerous returning true to roll the current controller we are checking vs returning false, not rolling and waiting a better time to get the active controller? Could it be "forever" because of a bad state in the quorum?